ktor-server/ktor-server-core/jvm/src/io/ktor/http/content/StaticContent.kt

```diff
@@ -85,13 +113,32 @@ fun Route.files(folder: String) = files(File(folder))
  */
 fun Route.files(folder: File) {
     val dir = staticRootFolder.combine(folder)
+    val compressedTypes = staticContentEncodedTypes
```

Comment:
It is useful for the use-case where some assets are pre-compressed with different algorithms. For example:
```
static
\ -- js (compressed with gz)
\ -- css (compressed with br)
```
can be achieved with:
```
preCompressed(GZIP)
files("js")
preCompressed(BROTLI)
files("css")
```
Does it make sense? @cy6erGn0m 
I agree this needs to be documented

Reply:
Doesn't it makes more sense to have it structured, like 

```kotlin
preCompressed(GZIP) {
    files("js")
}
preCompressed(BROTLI) {
    files("css")
}
files("videos") // no compressed
```

Reply:
And if not, we definitely need a test for such case to ensure that BROTLI wins for CSS

Reply:
Hi @cy6erGn0m , regarding 
```
	files("xyz") // not precompressed
	preCompressed(GZIP)
	files("js") // gzip
	preCompressed(BROTLI)
	files("css") // brotli for this and the ones below
```
I added a test (not yet pushed) for it and it is working as expected.

About the dsl
```
	preCompressed(GZIP) {
	    files("js") // GZIP only
	}
```
I see it more of a configuration of the files, unlike enclosing `files` inside a `preCompressed`, which feels like hierarchical. One advantage it has, is that it would allow many `files()` inside a `preCompressed` block.

A possible disadvantage, is the resulting dsl in case another 'configuration' for the `files()` is needed in the future. In case it is also implemented as a block, then the code would look like:
```
	featureX {
		preCompressed(GZIP) {
	    	files("js") // GZIP only
		}
	}
	...
	featureY {
		featureX {
			preCompressed(GZIP) {
	    		files("js") // GZIP only
			}
		}
	}
```
An alternative implementation is having a configuration block for ```files()```. This would result in the following usage:
```
	files("js") {
	    preCompressed(GZIP) // GZIP only
	}
	files("css") {
	    preCompressed(BROTLI) // BROTLI only
	}
	files("videos") // not compressed
```
It would allow future features for ```files``` to be added without changing the dsl:
```
	files("js") {
	    preCompressed(GZIP) // GZIP only
	    featureX()
	}
	files("css") {
	    featureX()
	    featureY = 'abs'
	    preCompressed(BROTLI) // BROTLI only
	}
	files("videos") // not compressed
```
The disadvantage comes when the user has many folders using preCompressed files, it would look like:
```
	files("js") {
	    preCompressed(GZIP) // GZIP only
	}
	files("css") {
		// Again the same config
	    preCompressed(GZIP) // GZIP only
	}
```

What do you think? Should I implement either of those? Or could it be addressed and refactored in a future once a decision on this part of the DSL is made?

Reply:
This is why nesting "files" in "preCompressed" in better in the last example. 
Regarding the example with featureX and Y: there is a chance of having different config block types but it's unlikely because one wants to have all functions like "file", "files" and other to work inside so both featureX and featureY should have the same `Route` receiver and, therefore, no problem since we are using `Route` to extend as well.

The disadvantage of nesting comparing to the flat model is that it intoduces nesting that makes a little bit more "noize". But I believe it looks more obvious, eliminates questions and possible confusion.

Reply:
Thanks for the feedback @cy6erGn0m , I'll add the requested changes

Reply:
Updated with the requested changes